virtcontainers/kata_agent.go

```diff
@@ -754,6 +756,14 @@ func (k *kataAgent) startSandbox(sandbox *Sandbox) error {
 		// (resolv.conf, etc...) and potentially all container
 		// rootfs will reside.
 		if sandbox.config.HypervisorConfig.SharedFS == config.VirtioFS {
+			// If virtio-fs uses either of the two cache options 'auto, always',
+			// the guest directory can be mounted with option 'dax' allowing it to
+			// directly map contents from the host. When set to 'none', the mount
+			// options should not contain 'dax' lest the virtio-fs daemon crashing
+			// with an invalid address reference.
```

Comment:
@ganeshmaharaj Do we know why we get the invalid address reference?

Reply:
@amshinde cause the guest thinks the memory region is shared between the host daemon and the guest and expects the host to know memory mapping to that region but the host process was started with `cache=none` and does not understand the translation.